src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -567,13 +559,15 @@ protected void update() {
       @Override
       public Boolean call() throws Exception
       {
+         // Speculative increment of totalConnections with expectation of success
+         if (totalConnections.incrementAndGet() > config.getMaximumPoolSize()) {
+            totalConnections.decrementAndGet(); // Pool is at max size
+            return Boolean.FALSE;
+         }
```

Comment:
Nevermind, this is still a bad change.  Now the pool is going to sit, for example, with ``totalConnections=10`` even though no connections can be made to the database (because it is down).  That is incorrect.